### ISSUE TITLE
Hacktoberfest 2021 (OpenGenus / Cosmos)

### ISSUE THREAD

**Issue by AdiChat** [OpenGenus/cosmos#6077]
Welcome to **Hacktoberfest 2021**

We, strongly, suggest you to apply for [your **Internship Program**](http://internship.opengenus.org) and contribute to Hacktoberfest in parallel with help from our mentors.

---

To get started on your journey in Hacktoberfest 2021, follow these steps:

* Register at [Hacktoberfest site](https://hacktoberfest.digitalocean.com/register).
* Fork [Cosmos](https://github.com/OpenGenus/cosmos/) to get a local copy as your workspace
* Learn how to make a perfect pull request ([guide](https://iq.opengenus.org/create-a-pull-request-at-github/))
* Make a basic contribution like improving existing implementations, simple bugs (like #5891) and more to get familiar with the flow
* Make 4 quality contributions (like implementing an algorithm/ problem). You can open a [new issue](https://github.com/OpenGenus/cosmos/issues/new) mentioning the contribution you plan to make and open a pull request directly.

Some lists of algorithms to help you contribute:

* [List of Array problems](https://iq.opengenus.org/list-of-array-problems/)
* [List of Linked List problems](https://iq.opengenus.org/list-of-linked-list-problems/)
* [List of Binary Tree problems](https://iq.opengenus.org/list-of-binary-tree-problems/)
* [List of Graph Algorithms](https://iq.opengenus.org/list-of-graph-algorithms/)
* [List of Dynamic Programming problems](https://iq.opengenus.org/list-of-dynamic-programming-problems/)
* [List of Data Structures](https://iq.opengenus.org/list-of-advanced-data-structures/)

Some quality contributions:

* Implement new algorithms in language of your choice (Having strong implementation skills is useful)
* Write at OpenGenus IQ on a topic you are confident in or prepare questions on a given topic (Ability to explain ideas is important)
* Write at OpenGenus IQ on a new topic in the domain you are interested in (Approaching new topics correctly is the key to invention)
* Develop features for sister projects of Cosmos (challenging but interesting)

This helps you grow intellectually 🧠 and help you gain good opportunities. Search and ask our Interns how they have grown over the years.

---

To begin contributing, you shall get a quick overview of how Cosmos is used today. **Cosmos** is one of the most extensive code database and is used as:

* A datastore for various projects (such as [Quark](https://github.com/OpenGenus/quark)) for purposes such as offline searching
* Training ML models for applications such as identifying programming language
* Metadata for projects such as [OpenGenus IQ](https://iq.opengenus.org/), [Falcon](https://github.com/OpenGenus/falcon) and several others
* Offline Learning resource by several contributors

---

**Have a doubt or want us to review your contribution?** Ask us down 👇 and we will help you.

**Comment by Onkar-Birajdar** [comment]
hey hi,
What do I have to do to raise a pull request?
have to solve the above problem and submit the solution?

**Comment by AdiChat** [comment]
@Onkar-Birajdar You can browse through other open issues and contribute towards them or add an implementation of a new algorithm 

You can raise a pull request with your contribution. This is [the guide](https://iq.opengenus.org/create-a-pull-request-at-github/) to raise a pull request 👍
